orm/model_bucket_test.go

```diff
@@ -0,0 +1,174 @@
+package orm
+
+import (
+	"strconv"
+	"testing"
+
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest"
+	"github.com/iov-one/weave/weavetest/assert"
+)
+
+func TestModelBucket(t *testing.T) {
+	db := store.MemStore()
+
+	obj := NewSimpleObj(nil, &Counter{})
+	objBucket := NewBucket("cnts", obj)
+
+	b := NewModelBucket(objBucket)
+
+	if _, err := b.Put(db, []byte("c1"), &Counter{Count: 1}); err != nil {
+		t.Fatalf("cannot save counter instance: %s", err)
+	}
+
+	var c1 Counter
+	if err := b.One(db, []byte("c1"), &c1); err != nil {
+		t.Fatalf("cannot get c1 counter: %s", err)
+	}
+	if c1.Count != 1 {
+		t.Fatalf("unexpected counter state: %d", c1)
+	}
+
+	if err := b.Delete(db, []byte("c1")); err != nil {
+		t.Fatalf("cannot delete c1 counter: %s", err)
+	}
+	if err := b.Delete(db, []byte("unknown")); !errors.ErrNotFound.Is(err) {
+		t.Fatalf("unexpected error when deleting unexisting instance: %s", err)
+	}
+	if err := b.One(db, []byte("c1"), &c1); !errors.ErrNotFound.Is(err) {
+		t.Fatalf("unexpected error for an unknown model get: %s", err)
+	}
+}
+
+func TestModelBucketPutSequence(t *testing.T) {
+	db := store.MemStore()
+
+	obj := NewSimpleObj(nil, &Counter{})
+	objBucket := NewBucket("cnts", obj)
+
+	b := NewModelBucket(objBucket)
+
+	// Using a nil key should cause the sequence ID to be used.
+	if _, err := b.Put(db, nil, &Counter{Count: 111}); err != nil {
```

Comment:
What about testing the first (key) return value as well.
Rather than just assuming what it is (it should be opaque).

eg.

```
key, err := b.Put(...)
if err != nil {
....
}
...
```

Then change
 `if err := b.One(db, weavetest.SequenceID(1), &c1); err != nil {`
to
 `if err := b.One(db, key, &c1); err != nil {`
 
or at least `assert.Equal(t, weavetest.SequenceID(1), key)`

Reply:
I wanted to test that not giving the key is using the sequence. I will add the test to check if the key is equal to sequence ID 1.